app/lib/actions/katello/upstream_subscriptions/remove_entitlement.rb

```diff
@@ -0,0 +1,22 @@
+module Actions
+  module Katello
+    module UpstreamSubscriptions
+      class RemoveEntitlement < Actions::Base
+        middleware.use Actions::Middleware::KeepCurrentTaxonomies
+        middleware.use Actions::Middleware::PropagateCandlepinErrors
+
+        input_format do
+          param :entitlement_id
+        end
+
+        def run
+          output[:response] = ::Katello::Resources::Candlepin::UpstreamConsumer.remove_entitlement(input[:entitlement_id])
+        end
+
+        def run_progress
+          done? ? 1 : 0.1
```

Comment:
I should have clarified, the example I showed you isn't the typical way to weight a task. For that task, it would show as half done very quickly, as the first "half" of the task took a short amount of time, but the second "half" could take hours. This meant the progress bar would just jump to 50% and stay there a long time, which made me use this override.

I'm not sure if that fix is what you need (it very well could be), but there is also the run_progress_weight method, which we us [other places](https://git.io/vxzgg) which gives a task weight among all the other tasks.

Reply:
Thanks for clarifying @johnpmitsch - I think the way I've done it works fine (I did try run_progress too but it actually didn't seem to have an affect).

Reply:
must be a similar situation then! As a long term goal we should look at adding this functionality to dynflow since it seems to be common and we could possibly use in other places